### PR TITLE
TRUNK-5664 Replace User hbm mapping file with annotations

### PR DESCRIPTION
**Pull request by wluyima**
TRUNK-5664 added a dummy secondary users table

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5664

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by sherrif10** [comment]
Thanks @wluyima for this work, my work earlier had some issues with master branch,but i think it had no errors

File: api/src/main/java/org/openmrs/User.java [hunk 1]
```diff
@@ -35,30 +59,59 @@
  * key-value pairs for either quick info or display specific info that needs to be persisted (like
  * locale preferences, search options, etc)
  */
+@Entity
+@Table(name = "users")
+@Cacheable
+@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
+@SecondaryTable(name = "users_unused_fields")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/User.java]: Can we put some code comments explaining why we are introducing the users_unused_fields table?

Comment (wluyima) [reply api/src/main/java/org/openmrs/User.java]: Done

**Comment by sherrif10** [comment]
@wluyima, @dkayiwa@openmrs.org <dkayiwa@openmrs.org>  where was the problem
in my PR

**Comment by wluyima** [comment]
You can take a look at the merge commit and compare changes with your PR

**Pull request merged**